testutil/testutil.go

```diff
@@ -0,0 +1,98 @@
+package testutil
+
+import (
+	"net"
+	"sync"
+)
+
+// Test constants
+const (
+	CloseOnMsgIdx = 5
+	MaxMsgCount   = 10
+)
+
+// LocalTCPServer tcp server for testing specific network errors
+type LocalTCPServer struct {
+	lnmu     sync.RWMutex
+	MockConn *Conn
```

Comment:
That's a lot of should happening here. Feel free to chime in with specific pointers.....